modules/internal/String.chpl

```diff
@@ -1790,9 +1796,9 @@ module String {
         var myJoinedArray = "|".join(["a","10","d"]);
         writeln(myJoinedArray); // prints: "a|10|d"
 
-    :arg x: An array or tuple of :mod:`string <String>` values to be joined
+    :arg x: An array or tuple of :type:`string` values to be joined
 
-    :returns: A :mod:`string <String>`
+    :returns: A :type:`string`
```

Comment:
@e-kayrakli : If you'd be willing to review the changes I've made to the Bytes and String modules, I'd appreciate it.  I believe they reflect where we ended up in our discussion of different approaches, but would appreciate if you'd double-check me.